### PR TITLE
add more mount damavand examples

### PR DESCRIPTION
**Pull request by akaszynski**
Add more Mt. Damavand examles based on @banesullivan's work at https://banesullivan.com/pyvista/examples/damavand.html

Example files are already in the repo, we just need a download for it.

Includes a sector model example as well.

### CODE REVIEW

File: pyvista/examples/downloads.py [hunk 1]
```diff
@@ -4198,3 +4198,62 @@ def download_moonlanding_image(load=True):  # pragma: no cover
...
+    -------
+    pyvista.PolyData or str
+        DataSet or filename depending on ``load``.
+
+    Examples
+    --------
+    Download the Damavand dataset and plot it after warping it by its altitude.
+
+    >>> from pyvista import examples
+    >>> dataset = download_mount_damavand()
```
Comment (akaszynski) [inline pyvista/examples/downloads.py:4253]:
```suggestion
    >>> dataset = examples.download_mount_damavand()
```

**Comment by akaszynski** [comment]
This is ready for review.

File: pyvista/examples/downloads.py [hunk 2]
```diff
@@ -4198,3 +4198,62 @@ def download_moonlanding_image(load=True):  # pragma: no cover
...
+    """
+    return _download_and_read('AngularSector.vtk', load=load)
+
+
+def download_mount_damavand(load=True):  # pragma: no cover
+    """Download the Mount Damavand dataset.
+
+    Visualize 3D models of Damavand Volcano, Alborz, Iran. This is a 2D map
+    with the altitude embedded as ``'z'`` cell data within the
+    :class:`pyvista.Polydata`.
```
Comment (adeak) [inline pyvista/examples/downloads.py:4232]:
```suggestion
    :class:`pyvista.PolyData`.
```

**Review by adeak — APPROVED** [review]
LGTM, thanks.

**Review by banesullivan — APPROVED** [review]
Thanks!

**Pull request merged**